### DISCUSSION TITLE
OpenSSL exe encryption output is not complete

### DISCUSSION THREAD

**Discussion by pradeepviswav** [openssl/openssl#21638] [Q&A]
I'm facing problem in using OpenSSL.exe version 3.1.0 to encrypt and decrypt user entered password. I'm using the following command format for encryption and decryption.

Encryption:

openssl.exe enc -aes-256-cbc -pbkdf2 -md sha512 -iter 1000 -base64 -A -pass pass:abc@1234

Decryption:

openssl.exe enc -d -aes-256-cbc -pbkdf2 -md sha512 -A -iter 1000 -base64 -pass pass:U2FsdGVkX18+JRgpZj5qGfpy6f8R9VFPRndh+2L7tP4=

After the encryption command, only a part of the encrypted password is getting displayed. When we press Ctrl+C, rest of the content is displayed. When I used the full encrypted output in the decryption command, "error reading input file" error is displayed. I would like to know whether the command I'm using is correct or not.

**Comment by slontis** [comment]
You may want to read the doc/man1/openssl-enc.pod
The following example encrypts the file README.md using aes-256-cbc.

The key used for encryption is generated using PBKDF2 (which creates a key using a password as input)
````
>LD_LIBRARY_PATH=. ./apps/openssl enc -aes-256-cbc -pbkdf2 -md sha256 -iter 1000 -pass pass:abc@1234 -in README.md -out encrypted.bin
````
Decryption also requires the key, so it needs the password used by the encrypt.

````
>LD_LIBRARY_PATH=. ./apps/openssl enc -d -aes-256-cbc -pbkdf2 -md sha256 -iter 1000 -pass pass:abc@1234 -in encrypted.bin
````
